xesmf/frontend.py

```diff
@@ -409,23 +431,43 @@ def __call__(self, indata, keep_attrs=False):
 
         """
         if isinstance(indata, np.ndarray):
-            return self.regrid_numpy(indata)
+            return self.regrid_numpy(indata, skipna=skipna, na_thres=na_thres)
         elif isinstance(indata, dask_array_type):
-            return self.regrid_dask(indata)
+            return self.regrid_dask(indata, skipna=skipna, na_thres=na_thres)
         elif isinstance(indata, xr.DataArray):
-            return self.regrid_dataarray(indata, keep_attrs=keep_attrs)
+            return self.regrid_dataarray(
+                indata, keep_attrs=keep_attrs, skipna=skipna, na_thres=na_thres
+            )
         elif isinstance(indata, xr.Dataset):
-            return self.regrid_dataset(indata, keep_attrs=keep_attrs)
-        else:
-            raise TypeError(
-                'input must be numpy array, dask array, ' 'xarray DataArray or Dataset!'
+            return self.regrid_dataset(
+                indata, keep_attrs=keep_attrs, skipna=skipna, na_thres=na_thres
             )
+        else:
+            raise TypeError('input must be numpy array, dask array, xarray DataArray or Dataset!')
 
     @staticmethod
-    def _regrid_array(indata, *, weights, shape_in, shape_out, sequence_in):
+    def _regrid_array(indata, *, weights, shape_in, shape_out, sequence_in, skipna, na_thres):
+
         if sequence_in:
             indata = np.expand_dims(indata, axis=-2)
-        return apply_weights(weights, indata, shape_in, shape_out)
+
+        # skipna: set missing values to zero
+        if skipna:
+            missing = np.isnan(indata)
```

Comment:
it would be acceptable to merge into 1 line

Reply:
``missing`` is reused later!

Reply:
:+1: